### PR TITLE
Adding haskell and greek lambdas

### PR DESCRIPTION
**Pull request by donovansmith44**
In addition to your original request, I also added the explicit Haskell-style lambdas of (\x -> x) to the grammar. Let me know if you like it.

### CODE REVIEW

File: src/lambda_cli/core.clj [hunk 1]
```diff
@@ -16,21 +16,22 @@
              | LAMBDA VAR_EXP '.' L_EXP
              | '(' L_EXP ')'
              | L_EXP L_EXP
+             | '\\\\' VAR_EXP '->' L_EXP
```
Comment (brain-fuel) [inline src/lambda_cli/core.clj]: Might as well just change ' \ \ \ \ ' to LAMBDA. The original intent of my ask was just to make it easier to type in "lambda" in multiple forms. If we're going to have the '->' for separating the bound expression from the LAMBDA VAR_EXP we might as well allow it everywhere.

Comment (brain-fuel) [reply src/lambda_cli/core.clj:19]: Note: In my comment above, I actually typed the characters ' \ \ \ \ ' in order, but they got escaped.

**Review by brain-fuel — CHANGES_REQUESTED** [review]

**Review by donovansmith44 — COMMENTED** [review]
Allowing use of '->' operator everywhere.

**Review by donovansmith44 — COMMENTED** [review]
Allowed '->' to be used everywhere

**Review by donovansmith44 — COMMENTED** [review]
Allowed '->' to be used everywhere

**Review by donovansmith44 — COMMENTED** [review]
Def BINDing of abstractions of vars to L_EXPs

**Review by donovansmith44 — COMMENTED** [review]
I thought it might be easier to evaluate tree elements if we had keywords for construction rules, which allows for clearer bifurcation and descent.

File: src/lambda_cli/core.clj [hunk 2]
```diff
@@ -15,8 +15,12 @@
 (def lambda-calculus
   (insta/parser
    "L_EXP    = VAR_EXP
-             | LAMBDA VAR '.' L_EXP
-             | <'('> L_EXP L_EXP <')'>
+             | APPLY
+             | ABSTRACT
+             | <'('> L_EXP <')'>
```
Comment (brain-fuel) [inline src/lambda_cli/core.clj]: Correct with the apply and abstract bits. Technically (and I just looked this up) line 20 is not part of the lambda calculus grammar.

Comment (brain-fuel) [reply src/lambda_cli/core.clj]: Meaning this: <'('> L_EXP <')'>

**Review by brain-fuel — CHANGES_REQUESTED** [review]

File: src/lambda_cli/core.clj (hunk 2 above)
Comment (brain-fuel) [reply src/lambda_cli/core.clj]: Actually, I was wrong. 

Application is what I have for line 19 (in red)
Abstraction is LAMBDA VAR BIND L_EXP

**Review by donovansmith44 — COMMENTED** [review]
Fixing APPLY and L_EXP definitions

**Review by brain-fuel — APPROVED** [review]

**Pull request merged**